tensorflow/core/framework/types.h

```diff
@@ -68,9 +68,17 @@ class DeviceType {
 std::ostream& operator<<(std::ostream& os, const DeviceType& d);
 
 // Convenient constants that can be passed to a DeviceType constructor
-extern const char* const DEVICE_CPU;   // "CPU"
-extern const char* const DEVICE_GPU;   // "GPU"
-extern const char* const DEVICE_SYCL;  // "SYCL"
+#if !defined(TF_EXTERN)
```

Comment:
Like it. Want to call it TF_DLLEXPORT or TF_EXPORT ? Later might be nicer if you want to control visibility on linux.